### PR TITLE
fix #307

### LINKED ISSUE #307: VerifyWindowPoSt&VerifyWinningPoSt has memory leak
the  toFilPoStProofs function has cgo.NewPoStProof(pp, src[idx].ProofBytes). but it do't has free this new object

### PR DESCRIPTION
**Pull request by marco-storswift**

### CODE REVIEW

File: proofs.go [hunk 1]
```diff
@@ -813,18 +815,27 @@ func fromFilPoStProofs(src []cgo.PoStProofGo) ([]proof.PoStProof, error) {
...
 }
 
-func toFilPoStProofs(src []proof.PoStProof) ([]cgo.PoStProof, error) {
+func toFilPoStProofs(src []proof.PoStProof) ([]cgo.PoStProof, func(), error) {
 	out := make([]cgo.PoStProof, len(src))
 	for idx := range out {
 		pp, err := toFilRegisteredPoStProof(src[idx].PoStProof)
 		if err != nil {
-			return nil, err
+			makeCleanerPRO(out, idx)()
```
Comment (Stebalien) [inline proofs.go:823]:
```suggestion
			makeCleanerPOST(out, idx)()
```

**Comment by cryptonemo** [comment]
Has it been verified that this fixes the observed issue?

**Review by dignifiedquire — APPROVED** [review]
good catch, this should definitely be merged independently of if it fixes #307

**Comment by jennijuju** [comment]
@marco-storswift  thank you for the PR, (again) 💙

**Pull request merged**